### PR TITLE
add: langchain (+openapi schema)

### PR DESCRIPTION
**Pull request by YYYasin19**
Hi, this adds [`langchain`](https://github.com/hwchase17/langchain) to conda-forge.
The recipe was generated by `grayskull` and I was able to get a successful build with `mambabuild` locally.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by YYYasin19** [comment]
Apparently, `langchain-server` needs docker to be installed.
Since this functionality is optional, I excluded it from the tests.
There is no way to show the user any messages/instructions after installation, right? Like how `brew` does.

Also: There is a large number of packages under `run_constrained` (which makes sense for this package). However, that might be hard to keep up to in the future. Should I simply recall grayskull periodically?

**Comment by YYYasin19** [comment]
@conda-forge/python, would be nice to get some feedback :)

File: recipes/langchain/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    - sqlalchemy >=1.0.0,<2.0.0
+    - requests >=2.0.0,<3.0.0
+    - pyyaml >=5.4.1
+    - numpy >=1.0.0,<2.0.0
+    - openapi-schema-pydantic >=1.2.0,<2.0.0
+    - dataclasses-json >=0.5.7,<0.6.0
+    - tenacity >=8.1.0,<9.0.0
+    - aiohttp >=3.8.3,<4.0.0
+    - async-timeout >=4.0.0,<5.0.0
+  run_constrained:
```
Comment (ocefpaf) [inline recipes/langchain/meta.yaml:36]: I believe these are generated by grayskull based on the package extras/optional packages. It is super nice to have them but we need to keep them updated later! Be careful to always re-generate the recipe with grayskull when updating it on the feedstock.

File: recipes/langchain/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    - poetry-core
+    - pip
+  run:
+    - python >=3.8.1,<4.0
+    - pydantic >=1.0.0,<2.0.0
+    - sqlalchemy >=1.0.0,<2.0.0
+    - requests >=2.0.0,<3.0.0
+    - pyyaml >=5.4.1
+    - numpy >=1.0.0,<2.0.0
+    - openapi-schema-pydantic >=1.2.0,<2.0.0
```
Comment (ocefpaf) [inline recipes/langchain/meta.yaml:31]: There will be a race condition and the creation of this feedstock may fail b/c openapi-schema-pydantic is not in the channel yet, be sure to restart the CIs when openapi-schema-pydantic lands.

**Comment by ocefpaf** [comment]
> Also: There is a large number of packages under `run_constrained` (which makes sense for this package). However, that might be hard to keep up to in the future. Should I simply recall grayskull periodically?

Yep. You can also try adding:

```
bot:
  inspection: update-grayskull
```

to the conda-forge.yml in the feedstock.

**Comment by YYYasin19** [comment]
Nice, thanks for merging and all the pointers, will have a look!

**Pull request merged**